typing_tests/merge-pine-options.ts

```diff
@@ -0,0 +1,497 @@
+/// <reference types="node" />
+/* eslint-disable @typescript-eslint/no-empty-object-type */
+import { type Equals, EqualsTrue } from './utils';
+import type * as BalenaSdk from '..';
+import { mergePineOptions } from '../es2017/util';
+
+const sdk: BalenaSdk.BalenaSDK = {} as any;
+
+export let aAny: any;
+export let aNumber: number;
+export let anUndefined: undefined;
+export let aNumberOrUndefined: number | undefined;
+export let aStringOrNull: string | null;
+export let aString: string;
+export let anArray: any[];
+
+// $select properties
+await (async () => {
+	const noProps = await sdk.models.apiKey.getAll();
+
+	aNumber = noProps[0].id;
+	aString = noProps[0].created_at;
+	aStringOrNull = noProps[0].expiry_date;
+	aStringOrNull = noProps[0].description;
+	aStringOrNull = noProps[0].name;
+	aNumber = noProps[0].is_of__actor.__id;
+
+	const narrowDownSelectString = await sdk.models.apiKey.getAll({
+		$select: 'id',
+	});
+	aNumber = narrowDownSelectString[0].id;
+	// @ts-expect-error - test case
+	aString = narrowDownSelectString[0].created_at;
```

Comment:
For the cases that you want to make sure that the error is originating from the right side of the expression, you might better use `aAny = ...` (or a `console.log` I guess, but I think I prefer the aAny)